packages/server/src/core/router.ts

```diff
@@ -41,48 +36,32 @@ export interface ProcedureStructure {
 }
 
 export interface RouterDef<
-  // FIXME this should use RootConfig
-  TContext,
-  TErrorShape extends TRPCErrorShape<number>,
-  TMeta extends Record<string, unknown>,
+  TConfig extends AnyRootConfig,
   TRecord extends ProcedureRouterRecord,
 > {
+  _config: TConfig;
   router: true;
-  /**
-   * @internal
-   */
-  _ctx: TContext;
-  /**
-   * @internal
-   */
-  _errorShape: TErrorShape;
-  /**
-   * @internal
-   */
-  _meta: TMeta;
-  errorFormatter: ErrorFormatter<TContext, TErrorShape>;
-  transformer: CombinedDataTransformer;
   procedures: TRecord;
   record: TRecord;
   /**
    * V9 queries
    * @deprecated
    */
-  queries: {};
+  queries: any;
   /**
    * V9 mutations
    * @deprecated
    */
-  mutations: {};
+  mutations: any;
   /**
    * V9 subscriptions
    * @deprecated
    */
-  subscriptions: {};
-  isDev: boolean;
+  subscriptions: any;
```

Comment:
I think the example is erroring because of these

Reply:
yeah, but not having it broke other stuff

Reply:
i wanted the type `AnyRouter` and it's "children" to be _somewhat_ typesafe so we got autocompletion/typesafety when using 

```ts
const router: AnyRouter = null as any;

router._def._config
//                          ^  i want autocompletion etc here
```